freelist.go

```diff
@@ -326,3 +327,44 @@ func (f *freelist) reindex() {
 		}
 	}
 }
+
+func (f *freelist) addReadonlyTXID(tid common.Txid) {
+	f.readonlyTXIDs = append(f.readonlyTXIDs, tid)
+}
+
+func (f *freelist) removeReadonlyTXID(tid common.Txid) {
+	for i := range f.readonlyTXIDs {
+		if f.readonlyTXIDs[i] == tid {
+			last := len(f.readonlyTXIDs) - 1
+			f.readonlyTXIDs[i] = f.readonlyTXIDs[last]
+			f.readonlyTXIDs = f.readonlyTXIDs[:last]
+			break
```

Comment:
there's `slices.Delete()` which we can use here too. I'm wondering if it's better to have a map here too, so we can also panic on duplicated pages. This removal would just delete the first occurrence.

Reply:
> I'm wondering if it's better to have a map here too

It's trade-off. The `freePages` needs to sort before use. If I understand it correctly, `freePages` needs to allocate memory for sort.

Reply:
> there's `slices.Delete()` which we can use.

`slices.Delete` is O(len(s)-i) and it copes/moves all elements after the removed element, while our implementation is O(1).

> I'm wondering if it's better to have a map here too, so we can also panic on duplicated transactions.

No, we can't. Multiple readonly TXNs may have the same txid, and it's expected.